### DISCUSSION TITLE
OSGI HTTP Whiteboard instead of Spring?

### DISCUSSION THREAD

**Discussion by hel-ar** [epsilonlabs/picto-web#23] [Q&A]
Hi,

I was wondering how closely intertwined the spring framework is with Picto-web. We have a OSGi application where we use the OSGi HTTP Whiteboard service to expose JAXRS resources. I'd be curious if something if you think something like that would be possible to integrate with Picto Web?

**Comment by agarciadom** [comment]
A quick search of "spring" over our codebase mostly points to the PictoController and some WebSockets logic. In theory it should be possible to refactor this code to expose it in a different way, so long as the OSGi HTTP Whiteboard service supports WebSockets as well.

The other aspect is that this would require a different build system (Tycho) instead of the plain Maven build system that we're using at the moment. That would require a reorganisation of the repository.
